types/react-router/index.d.ts

```diff
@@ -164,3 +165,36 @@ export function useRouteMatch<Params extends { [K in keyof Params]?: string } =
 export function useRouteMatch<Params extends { [K in keyof Params]?: string } = {}>(
     path: string | string[] | RouteProps,
 ): match<Params> | null;
+
+export interface RouteRenderConfig {
+    path: string;
+    element?: React.ReactNode;
+    children: RouteConfig[];
+}
+
+export interface RouteRenderWithChildrenConfig {
+    path: string;
+    element: React.ReactNode;
+}
+
+export interface RouteRedirectConfig {
+    path: string;
```

Comment:
Path is common part of all configs. Please make it as a base Config and make others extending it.